enhancements/20240612-node-disk-manager-refactor.md

```diff
@@ -0,0 +1,67 @@
+# Node Disk Manager Refactor
+
+## Summary
+
+The Node Disk Manager is responsible for managing the raw device on the node. The current implementation is not flexible enough to support various storage backends. This enhancement is to refactor the Node Disk Manager to make it more flexible and extensible.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/6014
+
+## Motivation
+
+We need to refactor the current implementation to make the Node Disk Manager support various storage backends. This could help define the storage provisioner easily.
```

Comment:
I'd suggest that you could mention current limitation about how difficult if we'd like to add LVM and v2 provisioner, then it becomes our motivation to provide this proposal.

For example: 

If we don't use abstract concept to decouple current implementation, it'd be hard to add new provisioner.

Reply:
Thanks @Yu-Jack,
I also rewrote this part.
Hope that could give more clarity to the motivation!
Please help to recheck, thanks!